jest.config.js

```diff
@@ -3,13 +3,13 @@ const baseConfig = require('./jest.config.base');
 module.exports = {
   ...baseConfig,
   projects: [
-    // Our legacy backend tests, slowly being migrated to microservices
-    '<rootDir>/test/jest.config.js',
+    // Satellite
+    '<rootDir>/src/satellite/jest.config.js',
     // Our new microservices
     '<rootDir>/src/api/**/jest.config.js',
```

Comment:
If we change this to

```
    // Our new microservices
    '<rootDir>/src/api/dependency-discovery/jest.config.js',
    '<rootDir>/src/api/feed-discovery/jest.config.js',
    '<rootDir>/src/api/image/jest.config.js',
    '<rootDir>/src/api/parser/jest.config.js',
    '<rootDir>/src/api/posts/jest.config.js',
    '<rootDir>/src/api/search/jest.config.js',
    '<rootDir>/src/api/sso/jest.config.js',
```

`pnpm test satellite` will only run once. I think it's because we have a number of services with no tests and jest is tripping out

Reply:
Planet and Status do not have tests and Parser has a jest.config.js file but no tests

Reply:
`pnpm coverage` in root yields:

```
----------------------------------|---------|----------|---------|---------|-----------------------------------------------------------------------
File                              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------------------------------|---------|----------|---------|---------|-----------------------------------------------------------------------
All files                         |   77.49 |    65.32 |    69.1 |   77.62 |                                                                       
 src/api/dependency-discovery/src |   58.33 |        0 |      50 |   58.33 |                                                                       
  index.js                        |     100 |      100 |     100 |     100 |                                                                       
  router.js                       |   47.36 |        0 |      50 |   47.36 | 11,20-32
 src/api/feed-discovery/src       |     100 |      100 |     100 |     100 | 
  index.js                        |     100 |      100 |     100 |     100 | 
  middleware.js                   |     100 |      100 |     100 |     100 | 
  router.js                       |     100 |      100 |     100 |     100 | 
 src/api/image/src                |     100 |      100 |     100 |     100 | 
  index.js                        |     100 |      100 |     100 |     100 | 
 src/api/image/src/lib            |   73.33 |    55.55 |   66.66 |   71.79 | 
  image.js                        |     100 |      100 |     100 |     100 | 
  photos.js                       |   55.55 |        0 |   42.85 |      56 | 14-26
 src/api/image/src/routes         |   90.19 |    88.88 |   77.77 |      90 | 
  gallery.js                      |     100 |      100 |     100 |     100 |                                                                       
  image.js                        |   87.17 |    88.88 |   71.42 |   86.84 | 54-55,91,103-104
 src/api/posts/src                |   81.51 |       55 |   72.72 |   81.08 | 
  index.js                        |   88.88 |      100 |       0 |   88.88 | 9
  queue.js                        |      80 |    33.33 |     100 |      80 | 13-15
  redis.js                        |     100 |      100 |     100 |     100 | 
  storage.js                      |   72.46 |       20 |   65.62 |   71.42 | 35-36,77,103,107,120-139,143,191,203-204
  text-parser.js                  |     100 |       50 |     100 |     100 | 4-15
  validation.js                   |     100 |      100 |     100 |     100 | 
 src/api/posts/src/data           |   56.52 |    73.33 |   43.33 |   57.95 | 
  feed.js                         |   41.17 |       75 |   28.57 |   42.55 | 26,29,57-73,89,106-140,152,164-196
  post.js                         |    75.6 |    72.22 |   77.77 |    75.6 | 21-25,53,71,96-118
 src/api/posts/src/routes         |   46.97 |       36 |   38.09 |   46.97 | 
  feeds.js                        |   30.47 |     3.57 |   27.77 |   30.47 | 14-25,37-38,54-55,72-73,78-92,101-119,130-150,160-168,180-197,207-221
  posts.js                        |   86.36 |    77.27 |     100 |   86.36 | 42-43,96-104
 src/api/search/src               |     100 |    83.33 |     100 |     100 | 
  index.js                        |     100 |      100 |     100 |     100 | 
  search.js                       |     100 |    83.33 |     100 |     100 | 1,38-49,140-141                                                       
 src/api/search/src/bin           |     100 |      100 |     100 |     100 | 
  validation.js                   |     100 |      100 |     100 |     100 | 
 src/api/search/src/routes        |     100 |      100 |     100 |     100 | 
  query.js                        |     100 |      100 |     100 |     100 | 
 src/api/sso/src                  |   95.16 |    93.33 |   92.85 |   96.42 | 
  roles.js                        |     100 |      100 |     100 |     100 | 
  token.js                        |     100 |       75 |     100 |     100 | 50
  user.js                         |   94.59 |    95.83 |      95 |   94.44 | 149-150
  util.js                         |    87.5 |      100 |   66.66 |     100 | 
 src/backend/data                 |   92.82 |    90.32 |   85.18 |   94.47 | 
  admin.js                        |   81.81 |       25 |      60 |      90 | 30
  article-error.js                |   83.33 |       50 |     100 |   83.33 | 8
  feed.js                         |   96.22 |    91.66 |   95.45 |      96 | 32,89
  hash.js                         |     100 |      100 |     100 |     100 |                                                                       
  post.js                         |   98.76 |      100 |     100 |   98.66 | 58
  stats.js                        |   71.42 |       75 |      50 |      80 | 14,25,35-38
  user.js                         |     100 |      100 |     100 |     100 | 
 src/backend/feed                 |   57.29 |     41.5 |   85.71 |   57.29 | 
  processor.js                    |   54.32 |    44.89 |     100 |   54.32 | 31-41,70-71,76-77,85,94,115,129,135-136,139-140,147-171,213-221      
  queue.js                        |   73.33 |        0 |       0 |   73.33 | 20-36
 src/backend/lib                  |   62.82 |       44 |      30 |   64.47 | 
  config.js                       |   44.44 |    16.66 |     100 |   44.44 | 8-12
  elastic.js                      |   77.27 |       50 |      50 |   80.95 | 23,45-47
  queue.js                        |   46.42 |    28.57 |   21.42 |   46.42 | 33-73,81-95
  redis.js                        |   78.94 |       75 |      50 |   83.33 | 22-24
 src/backend/utils                |   71.59 |    50.87 |   64.28 |      71 | 
  indexer.js                      |   32.65 |        0 |       0 |   33.33 | 22-35,44-51,61-66,71-76,81-84,99-123,138-166                          
  logger.js                       |   79.16 |    61.53 |     100 |   79.16 | 25,35,47-48,52
  storage.js                      |    87.5 |      100 |    90.9 |   86.04 | 66-68,86-87,147
  text-parser.js                  |     100 |       50 |     100 |     100 | 9
  url-parser.js                   |     100 |      100 |     100 |     100 | 
  wiki-feed-parser.js             |   86.36 |       80 |     100 |   86.36 | 24-25,45-46,62,96
 src/backend/utils/html           |   91.33 |    82.69 |     100 |   91.26 | 
  dom.js                          |   81.81 |       50 |     100 |   81.81 | 15,21
  fix-iframe-width.js             |   88.88 |       80 |     100 |   88.88 | 19
  index.js                        |   95.65 |       75 |     100 |   95.65 | 20
  lazy-load.js                    |      80 |       80 |     100 |      80 | 7
  modify-pre.js                   |   91.66 |    88.88 |     100 |    90.9 | 16
  remove-empty-anchor.js          |   85.71 |    85.71 |     100 |   85.71 | 3
  remove-empty-paragraphs.js      |   88.88 |    88.88 |     100 |   88.88 | 5
  sanitize.js                     |    87.5 |    83.33 |     100 |    87.5 | 69,79
  syntax-highlight.js             |   97.14 |       80 |     100 |   97.14 | 40
 src/backend/web                  |   59.45 |     27.5 |   37.03 |   59.09 | 
  app.js                          |   77.14 |       20 |       0 |   77.14 | 77-85,96-97
  authentication.js               |   37.73 |    29.16 |   26.66 |   37.73 | 38,70,79-82,100-142,160-226,248,265
  validation.js                   |    82.6 |    33.33 |      60 |   81.81 | 49-52,101-102                                                         
 src/backend/web/routes           |   75.36 |    55.26 |   63.33 |   75.27 | 
  admin.js                        |   53.57 |        0 |       0 |   53.57 | 19-37
  auth.js                         |      44 |        0 |       0 |      44 | 16-21,37-38,48-61
  feed.js                         |   85.71 |    66.66 |      75 |   86.53 | 53-54,59,118-119,150-151
  feeds.js                        |    78.2 |    81.25 |     100 |    78.2 | 17-18,46-47,65-66,76,86-87,96-97,107,115-116,125,130-131
  health.js                       |     100 |      100 |     100 |     100 | 
  index.js                        |   88.88 |       25 |     100 |   88.88 | 38-50
  query.js                        |   83.33 |      100 |     100 |   83.33 | 14-15
  service-provider.js             |   71.42 |      100 |       0 |   71.42 | 15-16
  stats.js                        |   70.58 |      100 |      50 |   68.75 | 9-14
  user.js                         |   76.47 |       50 |     100 |   76.47 | 11-12,25-26
 src/satellite/src                |   89.61 |    81.69 |   70.96 |    90.5 | 
  app.js                          |     100 |     90.9 |     100 |     100 | 10
  create-error.js                 |     100 |      100 |     100 |     100 |
  elastic.js                      |   73.91 |       80 |      40 |   77.27 | 35,46-51
  hash.js                         |     100 |      100 |     100 |     100 |
  index.js                        |     100 |      100 |     100 |     100 |
  logger.js                       |      80 |       50 |     100 |      80 | 7
  middleware.js                   |      90 |    71.42 |     100 |      90 | 64-70
  redis.js                        |      75 |       75 |      25 |   78.94 | 33-38
  satellite.js                    |   88.23 |    82.35 |      70 |   88.23 | 12,27,58-59
  service-token.js                |     100 |      100 |     100 |     100 |
 src/web/src/components           |     100 |    83.33 |     100 |     100 |
  AboutFooter.tsx                 |     100 |    83.33 |     100 |     100 | 92
  Logo.tsx                        |     100 |      100 |     100 |     100 |
 test                             |    90.9 |    77.77 |   76.19 |   93.87 |
  fixtures.js                     |    90.9 |    77.77 |   76.19 |   93.87 | 95,161,165
 test/lib                         |     100 |      100 |     100 |     100 |
  authentication.js               |     100 |      100 |     100 |     100 |
----------------------------------|---------|----------|---------|---------|-----------------------------------------------------------------------
Test Suites: 43 passed, 43 total
Tests:       399 passed, 399 total
Snapshots:   2 passed, 2 total
Time:        49.627 s
Ran all test suites in 9 projects.
```